library/core/src/any.rs

```diff
@@ -582,6 +583,74 @@ impl dyn Any + Send + Sync {
 // TypeID and its methods
 ///////////////////////////////////////////////////////////////////////////////
 
+extern "C" {
+    type OpaqueStrBytes;
+}
+
+// NOTE(eddyb) these are needed because `extern { type }`s don't implement any
+// auto traits, and their absence ends up propagating to `TypeId` otherwise.
+unsafe impl Send for OpaqueStrBytes {}
+unsafe impl Sync for OpaqueStrBytes {}
+impl crate::panic::RefUnwindSafe for OpaqueStrBytes {}
```

Comment:
Does this need `UnwindSafe` and `Unpin`?

Reply:
My understanding is that `&T: UnwindSafe` is given by `T: RefUnwindSafe` and `&T: Unpin` is always true (`OpaqueStrBytes` is `!Sized` and always behind a reference).